src/registry.ts

```diff
@@ -223,7 +227,7 @@ export async function createRegistry(
 			const url = `${registryBaseUrl}${image.path}/blobs/uploads/${parameters.size > 0 ? "?" + parameters : ""}`;
 			const options: https.RequestOptions = URL.parse(url);
 			options.method = "POST";
-			if (auth) options.headers = { authorization: auth };
+			if (token) options.headers = { authorization: token };
```

Comment:
This line is the fix. I took the liberty to also refactor the GitHub (anonymous) and GitLab token fetching into their own functions.